### DISCUSSION TITLE
Resolved: Hex stuff

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#2476] [General]
The current hex stuff I've done is pretty wild and is all entangled. I imagine it is going to be a nightmare to be able to ack things.

To help showcase everything I put all the patches on https://github.com/rust-bitcoin/hex-conservative/pull/64 and then did https://github.com/rust-bitcoin/rust-bitcoin/pull/2473 to prove that it all works.

I'm open to any suggestions that will make the review/merge process easier. FTR the only PR that I think is contentious is the backwards stuff in https://github.com/rust-bitcoin/hex-conservative/pull/67

~I can't actually remember right at this minute how I got to thinking that we need a new release of `hex` for the next `rust-bitcoin` release, I'll post that if/when I remember or find the `rust-bitcoin` work that requires it.~

We can't do https://github.com/rust-bitcoin/rust-bitcoin/pull/2474 without fixing `hex` because of `bitcoin::string`. So unless we are willing to release `units` with just the `amounts` module then the `rust-bitcoin` release is blocked on all this hex stuff.

**Comment by apoelstra** [comment]
I think your individual changes are fairly clearly motivated. Thanks for building a tree of PRs like this but I think I'm content to review the non-draft ones (currently #67 and #71) as they come undrafted. Though probably I will take a look at #69 to understand the direction you're taking with the reversed-string stuff.

**Reply by tcharding** [reply]
Awesome, thanks.

**Comment by tcharding** [comment]
Stale.
